__tests__/functional/update-script.js

```diff
@@ -0,0 +1,456 @@
+
+      runScriptSyncAndExpectError({ testDir })
+    })
+
+    it('exits with error if package.json file contains string containing govuk-prototype-kit', () => {
+      const testDir = mktestDirSync('name-contains-govuk-prototype-kit')
+      fs.writeFileSync(path.join(testDir, 'package.json'), '{\n  "name": "govuk-prototype-kit-test"\n}')
+
+      runScriptSyncAndExpectError({ testDir })
+    })
+
+    it.skip('exits with error if folder does not contain a VERSION.txt file', () => {
+      const testDir = mktestDirSync('no-version-txt', { version_txt: false })
+      fs.mkdirSync(testDir)
+      fs.writeFileSync(path.join(testDir, 'package.json'), '{\n  "name": "govuk-prototype-kit"\n}')
+
+      runScriptSyncAndExpectError({ testDir })
+    })
+  })
+
+  describe('prepare', () => {
+    it('removes existing update folder', () => {
+      const testDir = 'prepare'
+      fs.mkdirSync(path.join(testDir, 'update'), { recursive: true })
+      const oldStat = fs.statSync(path.join(testDir, 'update'))
+
+      runScriptSyncAndExpectSuccess('prepare', { testDir })
+
+      const newStat = fs.statSync(path.join(testDir, 'update'))
+
+      // tests that update folder has not been replaced
+      expect(newStat.birthtimeMs).not.toBe(oldStat.birthtimeMs)
+    })
+
+    it('does not remove existing update folder if CLEAN=0 is set', () => {
+      const testDir = 'prepare-noclean'
+      fs.mkdirSync(path.join(testDir, 'update'), { recursive: true })
+      fs.writeFileSync(path.join(testDir, 'update', 'govuk-prototype-kit-empty.zip'), '')
+      const oldStat = fs.statSync(path.join(testDir, 'update'))
+
+      runScriptSyncAndExpectSuccess('prepare', { testDir, env: { CLEAN: '0' } })
+
+      const newStat = fs.statSync(path.join(testDir, 'update'))
+
+      // tests that update folder has not been replaced
+      expect(newStat.birthtimeMs).toBe(oldStat.birthtimeMs)
+    })
+
+    it('hides the update folder from git', () => {
+      const testDir = 'prepare-gitignore'
+      fs.mkdirSync(testDir)
+
+      runScriptSyncAndExpectSuccess('prepare', { testDir })
+
+      expect(
+        fs.readFileSync(path.join(testDir, 'update', '.gitignore'), 'utf8').trim()
+      ).toBe('*')
+    })
+  })
+
+  describe('fetch', () => {
+    it('downloads the latest release of the prototype kit into the update folder', () => {
+      const testDir = 'fetch'
+      fs.mkdirSync(path.join(testDir, 'update'), { recursive: true })
+
+      runScriptSyncAndExpectSuccess('fetch', { testDir })
+
+      fs.accessSync(path.join(testDir, 'update'))
+      fs.accessSync(path.join(testDir, 'update', latestReleaseArchiveFilename))
+
+      /*
+      // compare downloaded file to fixture file
+      expect(
+        fs.readFileSync(path.join(testDir, 'update', latestReleaseArchiveFilename)).equals(
+          fs.readFileSync(getCurrentReleaseArchiveFixture())
+        )
+      ).toBe(true)
+      */
```

Comment:
FYI code commented out here - I don't think this test has an `expect` at the moment so may not be running

Reply:
Same as above, would be good to test, haven't got around to making it work 😅